server/config/store_config.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/typeutil"
+	"go.uber.org/zap"
+)
+
+var (
+	// default region max size is 144MB
+	defaultRegionMaxSize = uint64(144)
+	// default region split size is 96MB
+	defaultRegionSplitSize = uint64(96)
+	// default region max key is 144000
+	defaultRegionMaxKey = uint64(1440000)
+	// default region split key is 960000
+	defaultRegionSplitKey = uint64(960000)
+)
+
+// StoreConfigManager is used to manage the store config.
+type StoreConfigManager struct {
+	mu     sync.RWMutex
+	config *StoreConfig
+	client http.Client
+	schema string
+}
+
+// NewStoreConfigManager creates a new StoreConfigManager.
+func NewStoreConfigManager(config *SecurityConfig) *StoreConfigManager {
+	manager := &StoreConfigManager{
+		schema: "http",
+	}
+	if config == nil {
+		return manager
+	}
+	if cfg, err := config.ToTLSConfig(); err == nil && cfg != nil {
+		manager.client = http.Client{
+			Transport: &http.Transport{TLSClientConfig: cfg},
+		}
+		manager.schema = "https"
+	}
+	return manager
+}
+
+// StoreConfig is the config of store like TiKV.
+// generated by https://mholt.github.io/json-to-go/.
+// nolint
+type StoreConfig struct {
+	Coprocessor `json:"coprocessor"`
+}
+
+// Coprocessor is the config of coprocessor.
+type Coprocessor struct {
+	RegionMaxSize   string `json:"region-max-size"`
```

Comment:
how about using atomic value to avoid introducing lock?

Reply:
good idea.